### ISSUE TITLE
Slow

### ISSUE THREAD

**Issue by NTrukhanovets** [petkaantonov/bluebird#1294]
1) What version of bluebird is the issue happening on? 3.4.6

2) What platform and version? (For example Node.js 0.12 or Google Chrome 32) IE11

Promises work slowly in IE11. I have found this issue also, and as offered solution - use minified version of Bluebird. And yes, it helped, but could you, please, explain the core issue of this? And why it works fast with minified version?

https://github.com/systemjs/systemjs/issues/1220

Thank you!

**Comment by phpnode** [comment]
You almost certainly have long stack traces enabled which should be a development only feature because although they are incredibly useful during development they absolutely destroy promise performance. The minified build disables them.

**Comment by NTrukhanovets** [comment]
@phpnode, according to documentation, I can switch them off by setting environment variables:
```sh
NODE_ENV=production 
BLUEBIRD_LONG_STACK_TRACES=0 
BLUEBIRD_WARNINGS=0 
BLUEBIRD_DEBUG=0
```

I did it here: https://github.com/yankouskia/bluebird_slowness
Here is the example with webpack, where I have 2 ways to get bundle (production and development)
Also, I have html file `DIRECT_EXAMPLE.html`, in which I tried to include development and minified version of bluebird according to documentation here: http://bluebirdjs.com/docs/install.html

And I got next results:
![image](https://cloud.githubusercontent.com/assets/11259493/20922128/20a7b544-bbb8-11e6-8126-2714848b286c.png)
![image](https://cloud.githubusercontent.com/assets/11259493/20922137/23b57c9e-bbb8-11e6-9fc6-b56a398522c2.png)
![image](https://cloud.githubusercontent.com/assets/11259493/20922147/2697be22-bbb8-11e6-8c36-5ebcd5bb8369.png)
![image](https://cloud.githubusercontent.com/assets/11259493/20922152/29014642-bbb8-11e6-9fbd-eab7b415be6b.png)

Could you please, explain such result?
Thanks.

**Comment by petkaantonov** [comment]
If you downloaded the non-minified version, it has hard-coded long stack traces enabled ([see line 691](view-source:https://cdn.jsdelivr.net/bluebird/3.4.5/bluebird.js)). The downloads are meant to be used without any package manager with direct script tags, if you have package manager then those environment variables should work.

**Comment by yankouskia** [comment]
@petkaantonov 
there is function `env` in source code:
```js
function env(key, def) {
    return isNode ? process.env[key] : def;
}
```
`isNode` is false, because code is executed in browser, but global constants are configured at compile time. So I will never get true result.

Why do you need isNode here? I removed it and it worked correctly. Are there any reasons or I could create PR and fix it?

**Comment by deser** [comment]
And moreover I don't see why `def` is needed here as seems it isn't used anywhere

**Comment by yankouskia** [comment]
@petkaantonov , @benjamingr , @spion could anybody help with this, please? Or review PR? 
Thanks

**Comment by yankouskia** [comment]
@petkaantonov, @benjamingr, @spion,
guys, could you please merge fix for this issue?

**Comment by deser** [comment]
Guys?

**Comment by benjamingr** [comment]
None of us are going to merge this until we've asserted this works, I've tested it a bit and left comments but I'm still not sure - it seems to work.

I tested it last weekend, but a unit test would be nice. Also, @petkaantonov has been busy recently - he'll probably merge in the coming days.

**Comment by yankouskia** [comment]
@benjamingr , could you please publish to npm updated version of bluebird? Thanks!

**Comment by deser** [comment]
Guys? :) 
Can we expect this would be published as soon as possible?

**Comment by petkaantonov** [comment]
published 3.4.7

**Comment by yankouskia** [comment]
@petkaantonov , thank you!
